daemons/pacemakerd/pacemakerd.c

```diff
@@ -1394,7 +1466,13 @@ main(int argc, char **argv)
         mainloop_add_signal(SIGTERM, pcmk_shutdown);
         mainloop_add_signal(SIGINT, pcmk_shutdown);
 
-        init_children_processes();
+        if (running_with_sbd) {
+            pacemakerd_state = XML_PING_ATTR_PACEMAKERDSTATE_WAITPING;
+            startup_trigger = mainloop_add_trigger(G_PRIORITY_HIGH, init_children_processes, NULL);
+        } else {
+            pacemakerd_state = XML_PING_ATTR_PACEMAKERDSTATE_STARTINGDAEMONS;
+            init_children_processes(NULL);
+        }
```

Comment:
Have you traced what happens when pacemakerd itself respawns? find_and_track_existing_processes() will "adopt" any child daemons already running, so maybe we shouldn't wait for a ping in that case (the assumption would be that the ping succeeded with the previous pacemakerd instance), but I'm not sure.

Reply:
Was thinking about that but for simplicity and to prevent it from coming up when it shouldn't under some respawn-race I decided to wait this additional second.